manager/integration/tests/test_metric.py

```diff
@@ -158,6 +161,21 @@ def filter_metric_by_labels(metrics, labels):
         assert total_metrics["value"] >= 0.0
 
 
+def wait_for__metric_count_all_nodes(client, core_api, metric_name, metric_labels, expected_count): # NOQA
+    for _ in range(RETRY_COUNTS):
+        time.sleep(RETRY_INTERVAL)
+
+        try:
+            check_metric_count_all_nodes(client, core_api, metric_name,
+                                         metric_labels, expected_count)
+            break
+        except AssertionError:
+            continue
+
+    check_metric_count_all_nodes(client, core_api, metric_name,
```

Comment:
Why need to run this check again?

Reply:
We could also `raise AssertionError("Cannot find the metric matching the labels")`, but having another run would reveal the specific assertion error, I think this way gives a bit more clarity.
```
>       assert len(filtered_metrics) == expected_count
E       assert 1 == 4
E         -1
E         +4
```